docs/source/point-evaluation.rst

```diff
@@ -0,0 +1,116 @@
+.. only:: html
+
+  .. contents::
+
+Point evaluation
+================
+
+Firedrake can evaluate :py:class:`~.Function`\s at arbitrary physical
+points.  This feature can be useful for the evaluation of the result
+of a simulation.  Two APIs are offered to this feature: a
+Firedrake-specific one, and one from UFL.
+
+
+Firedrake API
+-------------
+
+Firedrake offers a convenient API for evaluating functions at
+arbitrary points via :func:`~.Function.at`:
+
+.. code-block:: python
+
+   # evaluate f at a 1-dimensional point
+   f.at(0.3)
+
+   # evaluate f at two 1-dimensional points, or at one 2-dimensional point
+   # (depending on f's geometric dimension)
+   f.at(0.2, 0.4)
+
+   # evaluate f at one 2-dimensional point
+   f.at([0.2, 0.4])
+
+   # evaluate f at two 2-dimensional point
+   f.at([0.2, 0.4], [1.2, 0.5])
+
+   # evaluate f at two 2-dimensional point (same as above)
+   f.at([[0.2, 0.4], [1.2, 0.5]])
+
+While in these examples we have only shown lists, other *iterables*
+such as tuples and ``numpy`` arrays are also accepted. The following
+are equivalent:
+
+.. code-block:: python
+
+   f.at(0.2, 0.4)
+   f.at((0.2, 0.4))
+   f.at([0.2, 0.4])
+   f.at(numpy.array([0.2, 0.4]))
+
+For a single point, the result is ``numpy`` array, or a tuple of
+``numpy`` arrays in case *mixed* functions.  When evaluating multiple
+points, the result is a list of values for each point.
```

Comment:
Yes, currently the second version happens. But I'm willing to do the first if @dham agrees.